library/blob.c

```diff
@@ -1099,7 +1172,17 @@ int eblob_blob_iterate(struct eblob_iterate_control *ctl)
 		}
 	}
 
-	err = eblob_blob_iterator(&iter_priv);
+	if (!(ctl->flags & EBLOB_ITERATE_FLAGS_BY_POSITION))
+		err = eblob_blob_iterator(&iter_priv);
+
+	if (ctl->flags & EBLOB_ITERATE_FLAGS_BY_POSITION) {
```

Comment:
maybe just `else` because the condition is the opposite one?

Reply:
done